### PR TITLE
demo(timepicker): decompose mousewheel-arrowkeys

### LINKED ISSUE #3546: docs(timepicker): decompose Mouse wheel and Arrow keys demo
need to decompose Mouse wheel and Arrow keys on two different demos(#3265): 
1) Mouse wheel
2) Arrow keys

### PR DESCRIPTION
**Pull request by tarusin**
decompose mousewheel-arrowkeys to mousewheel and arrow keys different demos

close #3546

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<p>Arrow keys disabled</p>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html]: IMO it's better for each of these demos to have a button that disables/enables arrow keys or mosuewheel

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<timepicker [(ngModel)]="myTime2" [arrowkeys]="false" [readonlyInput]="!isEnabled"></timepicker>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html]: readonly? initial idea was to disable / enable the possibility to use arrow keys or mouse wheel. So it should look like this `[arrowkeys]="allowArrowKeys"`

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+<hr>
+
+<button type="button" class="btn btn-info section bd-example" (click)="isEnabled=!isEnabled">Enable / Disable input</button>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html]: spaces
`(click)="isEnabled = !isEnabled"`

File: demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<timepicker [(ngModel)]="myTime1" [mousewheel]="false" [readonlyInput]="!isEnabled"></timepicker>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.html]: same, `[mousewheel]="allowMouseWheel"`

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<timepicker [(ngModel)]="myTime2" [arrowkeys]="allowArrowKeys"></timepicker>
+<hr>
+
+<button type="button" class="btn btn-info section bd-example" (click)="allowArrowKeys = !allowArrowKeys">Enable / Disable input</button>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.html]: `Enable / Disable input` should be changed to `Enable / Disable arrow keys`

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-timepicker-arrowkeys',
+  templateUrl: './arrowkeys.html'
+})
+export class DemoTimepickerArrowkeysComponent {
+  allowArrowKeys = true;
+  myTime2: Date = new Date();
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.ts]: change to `myTime`

File: demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<timepicker [(ngModel)]="myTime1" [mousewheel]="allowMouseWheel"></timepicker>
+<hr>
+
+<button type="button" class="btn btn-info section bd-example" (click)="allowMouseWheel = !allowMouseWheel">Enable / Disable input</button>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.html]: same

File: demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-timepicker-mousewheel',
+  templateUrl: './mousewheel.html'
+})
+export class DemoTimepickerMousewheelComponent {
+  allowMouseWheel = true;
+  myTime1: Date = new Date();
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.ts]: same

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Please, fix failed unit tests (maybe, rebase could help you)
unitfail
unitfaillog

File: demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-timepicker-arrowkeys',
+  templateUrl: './arrowkeys.html'
+})
+export class DemoTimepickerArrowkeysComponent {
+  allowArrowKeys = true;
+  myTime: Date = new Date();
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/arrowkeys/arrowkeys.ts]: remove type annotation 
if type annotation isn't provided, typescript will use the type of the first assigned value

File: demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-timepicker-mousewheel',
+  templateUrl: './mousewheel.html'
+})
+export class DemoTimepickerMousewheelComponent {
+  allowMouseWheel = true;
+  myTime: Date = new Date();
```
Comment (IlyaSurmay) [inline demo/src/app/components/+timepicker/demos/mousewheel/mousewheel.ts]: same

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — APPROVED** [review]

**Comment by YevheniiaMazur** [comment]
Tested, looks good

**Pull request merged**